src/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -41,35 +44,91 @@ internal struct ReadStackFrame
         public int PropertyIndex;
         public List<PropertyRef> PropertyRefCache;
 
-        // The current JSON data for a property does not match a given POCO, so ignore the property (recursively).
-        public bool Drain;
+        /// <summary>
+        /// Is the current object an Enumerable, Dictionary or IDictionaryConstructible.
+        /// </summary>
+        public bool IsProcessingCollectionObject()
+        {
+            return IsProcessingObject(ClassType.Enumerable | ClassType.Dictionary | ClassType.IDictionaryConstructible);
+        }
 
-        public bool IsCollectionForClass => IsEnumerable || IsDictionary || IsIDictionaryConstructible;
-        public bool IsCollectionForProperty => IsEnumerableProperty || IsDictionaryProperty || IsIDictionaryConstructibleProperty;
+        /// <summary>
+        /// Is the current property an Enumerable, Dictionary or IDictionaryConstructible.
+        /// </summary>
+        public bool IsProcessingCollectionProperty()
+        {
+            return IsProcessingProperty(ClassType.Enumerable | ClassType.Dictionary | ClassType.IDictionaryConstructible);
+        }
 
-        public bool IsIDictionaryConstructible => JsonClassInfo.ClassType == ClassType.IDictionaryConstructible;
-        public bool IsDictionary => JsonClassInfo.ClassType == ClassType.Dictionary;
+        /// <summary>
+        /// Is the current object or property an Enumerable, Dictionary or IDictionaryConstructible.
+        /// </summary>
+        public bool IsProcessingCollection()
+        {
+            return IsProcessingObject(ClassType.Enumerable | ClassType.Dictionary | ClassType.IDictionaryConstructible) ||
```

Comment:
Save the `ClassType` in a local. Seems to produce smaller asm (although it would be good to verify in VS with the actual usage, particularly with inlining):
[sharplab.io](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgJgIwCwAUFgMwAEu5AwuQN7HlOUVYAs5AsgBQCU9jZgF9BTUeQBSUAPYA7agBtoUAJKyAZtMkz5SqKo3S1sCLIDGcegHM4MANzkotuyKLNtcgAoJpAB0QwAJ5qmh6y3n4BwYbGMKYW1s6Ozq7u4mTkwNLSCuQqUBEW+gCWslYAIsVmMMVyEAiBAPIIKpXVtbL1gdRysAgAru3ACnB84gxu7sxYAOx5BT5FUKVWjcAAVnDV3IrKACqB/gB0bTV1DeQYNHpQB8etVWedDT2yfYM1w3D8GBjiU8x8oU4CUyhF/Aggjsbnc4CdHh0updrvtDnCHu1zt1ejABkMRrw7OJUsx0hQsjl5sDQRUEVjmhinl1Xu98XAAPrUCBmAAWcAAMtIzBAFGNJkwJgCmLt9LDyEF/OQALwo2Vo+GY56BZEy27qxmIl44vGfEZE8VTWZUxYg5ZlNabbYK76XP4WqXWoW2lbgqLcZ2E4niMmZbK5IE2mkOrYwaGoxW4WFQXjjf7TObcKRyXUhIxvOLmOG6uUAMkoOCT/AAhCqAAzm9yuEMU8MLL0032QwJxtUJito5Op93p8hVrPhHwQoK52LxdFtqeBTw5KraksltMAzM6TvTmL5udHYto8hlxMD6t1hvCYMWjJUcc5wxDj0ZY+K9+WOg2ezJewkpgAPIEMHx3ScolzF8pTfGET0/RJfycf9NxYUNKQjSIu2XBRVwQhwkJcIMLRDOBZH6MBVT1RUQEyQJ4CggEAHpGPlNFpHUbhpA2GMU2HJgAFVZAAa1kaQAHdZGVchawADzwJAUOY8hPEaahGnIaQEHIBAqh5cgABMIDiVj/BQ6Nqik2ScAUvjyCUgA1EV+ksTSNO4iyxOKGA9IgcgzDkAA3AJECOFDHIUZzLJktgbI9JSsDwUgUIAUTIsBEAgL4ooADliqUlINLEUNOQ1tTrOTazypiWPyAyQUQYoRWKAAvOB9No8hfGUFZyF8wqtXlLQvKgPzjQ+TTFJY4ojjhchijAMB+jiLL9LpZ5ihBJByAACWgHllpGLaAGVNPgfT+WKWAqqmfrmTGtkopwSqiKEIA===)

Reply:
The logical |s define constant at compile time. Creating a local makes the ASM have an extra MOV statement.